pallets/staking-coefficient-reward/src/migrations.rs

```diff
@@ -0,0 +1,43 @@
+//! Storage migrations for the parachain-staking  pallet.
+
+use super::*;
+use frame_support::{
+	dispatch::GetStorageVersion,
+	pallet_prelude::{StorageVersion, ValueQuery},
+	storage_alias,
+	traits::Get,
+	weights::Weight,
+};
+
+const CURRENT_STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
+
+pub(crate) fn on_runtime_upgrade<T: Config>() -> Weight {
+	upgrade::Migrate::<T>::on_runtime_upgrade()
+}
+
+mod upgrade {
+	use super::*;
+
+	#[storage_alias]
+	type CoefficientConfig<T: Config> = StorageValue<Pallet<T>, u8, ValueQuery>;
+
+	/// Migration implementation that renames storage HardCap into MaxCurrencySupply
+	pub struct Migrate<T>(sp_std::marker::PhantomData<T>);
+
+	impl<T: Config> Migrate<T> {
+		pub fn on_runtime_upgrade() -> Weight {
+			let mut weight_writes = 0;
+			let mut weight_reads = 0;
+			let onchain_storage_version = Pallet::<T>::on_chain_storage_version();
+			if onchain_storage_version.eq(&CURRENT_STORAGE_VERSION) {
```

Comment:
It looks like Pallet's on_chain_storage_version is 8, and we run migration logic when CURRENT_STORAGE_VERSION(which is 0) is equal to it. The migration will not run in this case, and CoefficientConfig will not be update. **I'm assuming this wasn't the intention,** so we should check for inequality(or 'less than' case if we want the migration to run.